booking/src/main/java/org/dcsa/conformance/standards/booking/checks/BookingChecks.java

```diff
@@ -0,0 +1,645 @@
+    (mav) -> {
+      mav.submitAllMatching("requestedEquipments.*.references.*.type");
+        mav.submitAllMatching("requestedEquipments.*.commodities.*.references.*.type");
+    },
+    JsonAttribute.matchedMustBeDatasetKeywordIfPresent(BookingDataSets.REFERENCE_TYPES)
+  );
+
+  private static final JsonContentCheck TLR_CC_T_COMBINATION_VALIDATIONS = JsonAttribute.allIndividualMatchesMustBeValid(
+    "Validate combination of 'countryCode' and 'type' in 'taxAndLegalReferences'",
+    (mav) -> {
+      mav.submitAllMatching("documentParties.*.party.taxLegalReferences.*");
+    },
+    JsonAttribute.combineAndValidateAgainstDataset(BookingDataSets.LTR_CC_T_COMBINATIONS, "countryCode", "type")
+  );
+
+  private static final JsonContentCheck ISO_EQUIPMENT_CODE_VALIDATION = JsonAttribute.allIndividualMatchesMustBeValid(
+    "Validate ISO Equipment code",
+    (mav) -> {
+      mav.submitAllMatching("requestedEquipments.*.ISOEquipmentCode");
+    },
+    JsonAttribute.matchedMustBeDatasetKeywordIfPresent(BookingDataSets.ISO_6346_CONTAINER_CODES)
+  );
+
+  private static final JsonContentCheck OUTER_PACKAGING_CODE_IS_VALID = JsonAttribute.allIndividualMatchesMustBeValid(
+    "Validate that 'packagingCode' is a known code",
+    (mav) -> mav.submitAllMatching("requestedEquipments.*.commodities.*.outerPackaging.packageCode"),
+    JsonAttribute.matchedMustBeDatasetKeywordIfPresent(BookingDataSets.OUTER_PACKAGING_CODE)
+  );
+
+  private static Consumer<MultiAttributeValidator> allDg(Consumer<MultiAttributeValidator.AttributePathBuilder> consumer) {
+    return (mav) -> consumer.accept(mav.path("requestedEquipments").all().path("commodities").all().path("outerPackaging").path("dangerousGoods").all());
+  }
+  private static final JsonContentCheck IS_EXPORT_DECLARATION_REFERENCE_PRESENT = JsonAttribute.ifThen(
+    "Check Export declaration reference ",
+    JsonAttribute.isTrue(JsonPointer.compile("/isExportDeclarationRequired")),
+    JsonAttribute.mustBePresent(JsonPointer.compile("/exportDeclarationReference"))
+  );
+
+  private static final JsonContentCheck IS_IMPORT_DECLARATION_REFERENCE_PRESENT = JsonAttribute.ifThen(
+    "Check Import declaration reference presence",
+    JsonAttribute.isTrue(JsonPointer.compile("/isImportLicenseRequired")),
+    JsonAttribute.mustBePresent(JsonPointer.compile("/importLicenseReference"))
+  );
+
+  private static final JsonContentCheck DOCUMENT_PARTY_FUNCTIONS_MUST_BE_UNIQUE = JsonAttribute.customValidator(
+    "Each document party can be used at most once",
+    JsonAttribute.path(
+      "documentParties",
+      JsonAttribute.unique("partyFunction")
+    ));
+
+  private static final JsonContentCheck COMMODITIES_SUBREFERENCE_UNIQUE = JsonAttribute.allIndividualMatchesMustBeValid(
+    "Each Subreference in commodities must be unique",
+    (mav) -> {
+      mav.submitAllMatching("requestedEquipments.*.commodities");
+    },
+    JsonAttribute.unique("subReference")
+  );
```

Comment:
Do we have a check that subreferences must be provided on the commodity when the booking is confirmed? (I cannot see one)